proposals/lip-0070.md

```diff
@@ -434,6 +434,28 @@ def getClaimableRewards(stakerAddress: Address) -> dict[TokenID, uint64]:
     return rewards
 ```
 
+#### getExpectedReward
```

Comment:
I think the name of this function `getExpectedReward` and the function `getValidatorBlockReward` is not sufficiently clear to distinguish them. My suggestion would be:

- `getExpectedSharedBlockReward` (function here)
- `getExpectedBlockReward` (function in LIP 71)

Reply:
I am also not sure whether reward or rewards is better, maybe we could ask a native speaker.

Reply:
Or, if we also expect the endpoint to be used for shared fees in the future, we could call it `getExpectedSharedRewards`.

Reply:
switched to:

- `getExpectedStakingBlockReward` in LIP 70: this way it is clear that  i) it is expected, not guaranteed ii) it has to do with staking and iii) it is only for one block.
- `getExpectedValidatorRewards` (as it was) in LIP 71: This way it is clear that i) it is expected, ii) it is for validators; and I use "rewards" instead of "reward" since it gives potential rewards for several different time intervals.

Reply:
As for fees: I don't see how we could estimate expected transition fees and did not bother much about it now. Open to discuss this if you want. Then, a suitable name would be `getExpectedStakingRewardsPerBlock` I would say.

Reply:
The name in LIP 71 sounds good.

Regarding the name in LIP 71, it was not changed in the files. Further, I would find "shared" as part of the name nicer than staking, as it makes it more clear that this is what is shared after commission. Further, I would remove "block" from the name, as it is not necessarily just for one block, it just totally depends on the input parameter `validatorRewards` of the function. For instance, you could use as input parameter the yearly rewards for a validator or, in case in a sidechain fees are also shared with stakers, the yearly collected fees by a validator. Basically, this function allows to compute for any shared amount coming from any source, how much a single staker receives.

Reply:
For LIP70 function name:

- The file is now updated.
 - I got confused, or course Block does not make sense.
 - I think "staking rewards" is a standard term used. But ok switched to shared since I have not a strong opinion here.

Reply:
Looks good.